frontend/src/metabase/sharing/components/testSetup.js

```diff
@@ -0,0 +1,126 @@
+import fetchMock from "fetch-mock";
+import { renderWithProviders } from "__support__/ui";
+import { mockSettings } from "__support__/settings";
+import { createMockState } from "metabase-types/store/mocks";
+import { setupEnterprisePlugins } from "__support__/enterprise";
+
+import {
+  createMockDashboard,
+  createMockActionDashboardCard,
+  createMockDashboardOrderedCard,
+  createMockUser,
+  createMockCard,
+  createMockTokenFeatures,
+} from "metabase-types/api/mocks";
+
+import SharingSidebar from "./SharingSidebar";
+
+export const dashcard = createMockDashboardOrderedCard({ name: "dashcard" });
+
+const actionDashcard = createMockActionDashboardCard({
+  id: 2,
+  name: "actionDashcard",
+});
+const linkDashcard = createMockActionDashboardCard({
+  id: 3,
+  name: "linkDashcard",
+  card: createMockCard({ display: "link" }),
+});
+
+export const user = createMockUser();
+
+const dashboard = createMockDashboard({
+  ordered_cards: [dashcard, actionDashcard, linkDashcard],
+  parameters: [
+    {
+      name: "ID",
+      slug: "id",
+      id: "abcd1234",
+      type: "id",
+      sectionId: "id",
+    },
+  ],
+});
+
+export function setup(
+  { email, slack, tokenFeatures = {} } = {
+    email: true,
+    slack: true,
+    tokenFeatures: {},
+  },
+) {
+  const channelData = { channels: {} };
+
+  if (email) {
+    channelData.channels.email = {
+      type: "email",
+      name: "Email",
+      allows_recipients: true,
+      recipients: ["user", "email"],
+      schedules: ["hourly"],
+      configured: true,
+    };
+  }
+
+  if (slack) {
+    channelData.channels.slack = {
+      type: "slack",
+      name: "Slack",
+      allows_recipients: false,
+      schedules: ["hourly"],
+      configured: true,
+      fields: [
+        {
+          name: "channel",
+          type: "select",
+          displayName: "Post to",
+          options: ["#general", "#random", "#alerts"],
+          required: true,
+        },
+      ],
+    };
+  }
+
+  fetchMock.get("path:/api/pulse/form_input", channelData);
+
+  fetchMock.get("path:/api/user", {
+    data: [user],
+  });
+
+  fetchMock.get(
+    { url: `path:/api/pulse`, query: { dashboard_id: dashboard.id } },
+    [],
+  );
+
+  fetchMock.post("path:/api/pulse/test", 200);
+
+  const features = createMockTokenFeatures(tokenFeatures);
+  const storeSettings = mockSettings({ "token-features": features });
+
+  if (tokenFeatures) {
```

Comment:
Technically you can have token features on OSS without a token (and in fact you do), that's why in the doc we have a separate flag for calling `setupEnterprisePlugins`, and 3 test suites instead of 2; i.e. an additional one for EE with empty token features